optimum/exporters/tflite/base.py

```diff
@@ -0,0 +1,288 @@
+        """
+        if hasattr(self._config, "use_cache"):
+            return {"use_cache": False}
+
+        return None
+
+    @property
+    @abstractmethod
+    def inputs(self) -> List[str]:
+        """
+        List containing the names of the inputs the exported model should take.
+
+        Returns:
+            `List[str]`: A list of input names.
+        """
+        raise NotImplementedError()
+
+    @property
+    def outputs(self) -> List[str]:
+        """
+        List containing the names of the outputs the exported model should have.
+
+        Returns:
+            `List[str]`: A list of output names.
+        """
+        return self._TASK_TO_COMMON_OUTPUTS[self.task]
+
+    def generate_dummy_inputs(self) -> Dict[str, "tf.Tensor"]:
+        """
+        Generates dummy inputs that the exported model should be able to process.
+        This method is actually used to determine the input specs that are needed for the export.
+
+        Returns:
+            `Dict[str, tf.Tensor]`: A dictionary mapping input names to dummy tensors.
+        """
+        dummy_inputs_generators = self._create_dummy_input_generator_classes()
+        dummy_inputs = {}
+
+        for input_name in self.inputs:
+            input_was_inserted = False
+            for dummy_input_gen in dummy_inputs_generators:
+                if dummy_input_gen.supports_input(input_name):
+                    dummy_inputs[input_name] = dummy_input_gen.generate(input_name, framework="tf")
+                    input_was_inserted = True
+                    break
+            if not input_was_inserted:
+                raise RuntimeError(
+                    f'Could not generate dummy inputs for "{input_name}". Try adding a proper dummy input generator '
+                    "to the model TFLite config."
+                )
+
+        return dummy_inputs
+
+    @property
+    def inputs_specs(self) -> List["TensorSpec"]:
+        """
+        List containing the input specs for each input name in self.inputs.
+
+        Returns:
+            `List[tf.TensorSpec]`: A list of tensor specs.
+        """
+        dummy_inputs = self.generate_dummy_inputs()
+        return [
+            tf.TensorSpec(dummy_input.shape, dtype=dummy_input.dtype, name=input_name)
+            for input_name, dummy_input in dummy_inputs.items()
+        ]
+
+    def model_to_signatures(
+        self, model: "TFPreTrainedModel", **model_kwargs: Any
+    ) -> Dict[str, "tf.types.experimental.ConcreteFunction"]:
+        """
+        Creates the signatures that will be used when exporting the model to a `tf.SavedModel`.
+        Each signature can be used to perform inference on the model for a given set of inputs.
+
+        Auto-encoder models have only one signature, decoder models can have two, one for the decoder without
+        caching, and one for the decoder with caching, seq2seq models can have three, and so on.
```

Comment:
* Does this function only exist to aid SavedModel creation? 
* Does `model.save("my_model")` (`model` is an instance of `TFPretrainedModel`) lead to any error? 
* If this helps, TFLite allows us to load specific signatures: https://www.tensorflow.org/lite/guide/signatures

Reply:
- Yes, basically it provides the signatures that will be used for the export of the model.
- I haven't tried, but I think this is equivalent to what we do.
- That's exactly what we use: we export a signature that can be used easily by the `tf.lite.Interpreter`. For decoder and seq2seq models, we will define multiple signatures.